### PR TITLE
Fix: Fetch columns-to-types mapping from the target DW when determining a type of the time column

### PR DESCRIPTION
**Pull request by izeigerman**

### CODE REVIEW

File: sqlmesh/core/engine_adapter/base.py [hunk 1]
```diff
@@ -657,6 +662,10 @@ def _insert_overwrite_by_condition(
                     )
                 )
             query = t.cast("Query", query_or_df)
+
+            if where and isinstance(query, exp.Subqueryable):
```
Comment (tobymao) [inline sqlmesh/core/engine_adapter/base.py:666]: put this check into the add_where_to_query method directly

File: sqlmesh/core/model/definition.py [hunk 2]
```diff
@@ -38,7 +38,6 @@
     print_exception,
     serialize_env,
 )
-from sqlmesh.utils.pandas import filter_df_by_timelike
```
Comment (tobymao) [inline sqlmesh/core/model/definition.py:41]: can you delete this method now?

Comment (izeigerman) [reply sqlmesh/core/model/definition.py:41]: seems like I can

Comment (tobymao) [reply sqlmesh/core/model/definition.py:41]: how do you filter pandas dataframes then?

Comment (izeigerman) [reply sqlmesh/core/model/definition.py:41]: when it's translated to sql

Comment (izeigerman) [reply sqlmesh/core/model/definition.py:41]: removed the function

**Review by tobymao — APPROVED** [review]

**Pull request merged**